### PR TITLE
new entity NotificationRecipients

### PR DESCRIPTION
**Pull request by pnovotny**
Added new entity `NotificationRecipients` for `/notification_recipients` endpoint.

This PR is required by https://github.com/SatelliteQE/robottelo/pull/14592

### CODE REVIEW

**Comment by pnovotny** [comment]
PRT of https://github.com/SatelliteQE/robottelo/pull/14592 has passed.

File: nailgun/entities.py [hunk 1]
```diff
@@ -8916,3 +8916,17 @@ def search(self, fields=None, query=None, filters=None):
...
+
+
+class NotificationRecipients(Entity, EntityReadMixin):
+    """A representation of /notification_recipients endpoint."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'notifications': entity_fields.ListField(),
+        }
+        self._meta = {
```
Comment (Gauravtalreja1) [inline nailgun/entities.py:8928]: Could you add a unit test for this entity class?

**Review by Griffin-Sullivan — APPROVED** [review]
Ack pending test

File: nailgun/entities.py (hunk 1 above)
Comment (pnovotny) [reply nailgun/entities.py:8928]: Unit test added and passing:
```
pytest tests/test_entities.py -k NotificationRecipientsTestCase
============================================================================================================= test session starts ==============================================================================================================
...
collected 163 items / 162 deselected / 1 selected                                                                                                                                                                                              

tests/test_entities.py::NotificationRecipientsTestCase::test_to_json PASSED                                                                                                                                                              [100%]

====================================================================================================== 1 passed, 162 deselected in 0.20s =======================================================================================================
```

**Comment by pnovotny** [comment]
Up-to-date PRT passed: https://github.com/SatelliteQE/robottelo/pull/14592#issuecomment-2038087410
Added unit test also passed: https://github.com/SatelliteQE/nailgun/pull/1118#discussion_r1552318300

**Review by ogajduse — APPROVED** [review]

**Review by pondrejk — APPROVED** [review]

**Pull request merged**